src/test/java/com/rocketden/main/service/RoomServiceTests.java

```diff
@@ -0,0 +1,102 @@
+package com.rocketden.main.service;
+
+import com.rocketden.main.dao.RoomRepository;
+import com.rocketden.main.dto.room.CreateRoomRequest;
+import com.rocketden.main.dto.room.CreateRoomResponse;
+import com.rocketden.main.dto.room.JoinRoomRequest;
+import com.rocketden.main.dto.room.JoinRoomResponse;
+import com.rocketden.main.model.Room;
+import org.junit.Before;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class RoomServiceTests {
+
+    @Mock
+    private RoomRepository repository;
+
+    @Spy
+    @InjectMocks
+    private RoomService service;
+
+    @Before
+    public void setUp() {
+        service = new RoomService(repository);
+    }
+
+    @Test
+    public void createRoomSuccess() {
+        // Verify create room request succeeds and returns correct response
+        CreateRoomRequest request = new CreateRoomRequest();
+
+        // Mock generateRoomId to return a custom room id
+        Mockito.doReturn("012345").when(service).generateRoomId();
+        CreateRoomResponse response = service.createRoom(request);
+
+        verify(repository).save(Mockito.any());
+        assertEquals(CreateRoomResponse.SUCCESS, response.getMessage());
+        assertEquals("012345", response.getRoomId());
+    }
+
+    @Test
+    public void joinRoomSuccess() {
+        // Verify join room request succeeds and returns correct response
+        String roomId = "012345";
+
+        JoinRoomRequest request = new JoinRoomRequest();
+        request.setPlayerName("rocket");
+        request.setRoomId(roomId);
+
+        Room room = new Room();
+        room.setRoomId(roomId);
+
+        // Mock repository to return room when called
+        Mockito.doReturn(room).when(repository).findRoomByRoomId(eq(roomId));
```

Comment:
Note on the following line: 

```java
Mockito.doReturn(room).when(repository).findRoomByRoomId(eq(roomId));
```
This makes it so that when `repository.findRoomByRoomId` is called with a string equal to `roomId`, it will return the `room` object that you defined a couple lines above

Reply:
This looks great, thanks for adding this.